### DISCUSSION TITLE
Is Play Assets Delivery packaging available now in Air sdk?

### DISCUSSION THREAD

**Discussion by mehmury** [airsdk/Adobe-Runtime-Support#1481] [Q&A]
I tested it with Adobe air sdk 33.1.1.713 to use Play Assets delivery.
I followed the relevant instructions to package it as stated in the release notes. However, when I send the aab file created after packaging to the Google Play Console, it reports that I have exceeded the 150 mb limit.
Is Play Assets Delivery packaging available now in Air sdk?
I will be happy if you provide information.

**Comment by ajwfrost** [comment]
Yes it should be all supported, most likely it's not picked up the folder that needs to be flagged as the separate asset pack.. if you open the aab file in a zip file manager, you should be able to see whether there is a separate asset pack in there.. (should have 'base' folder plus metadata, and then a separate folder for each asset pack each with assets.pb and the files themselves in an 'assets' folder)

In your app descriptor you should have a definition e.g.
`                <assetPack id="fast_follow_asset_pack" delivery="fast-follow" folder="ff_asset_pack"/>`
and then you'd need to have your assets in a folder called 'ff_asset_pack' and include this folder in your package...

**Reply by mehmury** [reply]
Thanks.
I insert the assest information into the xml definition
```
`<assetPacks>`
         <assetPack delivery="install-time" folder="AP_Images" id="Ap_Images"/>
       </assetPacks>
```
I added as
When I look inside the package, the files seem to have been added.
located in assets.pb
The folder Ap_Images is located and the additional swf files that I added to its content appear.
However, I compile with Adobe Animate as aab.
In this way, when I send it to google play console, I get the warning that you have exceeded 150 mb.
![1](https://user-images.githubusercontent.com/51246471/146973927-8bd4c1f1-ac7e-4934-867f-ecfcccb60a67.png)
![2](https://user-images.githubusercontent.com/51246471/146973930-9b4f05c6-b17a-46cc-a387-8bbe3effe7b4.png)
![3](https://user-images.githubusercontent.com/51246471/146973939-a34b2eb8-87c8-4852-adff-d55f7c36fa29.png)
![4](https://user-images.githubusercontent.com/51246471/146973944-e8920be0-3b50-45ee-86e2-7990f663ff36.png)

**Reply by ajwfrost** [reply]
Everything looks fine for that .. but it's clear that the files didn't get put in the right place!
Can you please just check whether it's case-sensitive i.e. if you try using `Ap_Images` as the folder name in your app descriptor XML?
thanks

**Reply by mehmury** [reply]
Thanks. Yes, xml file has capitalization error. Now upload is accepted.
Is there a method where I can get the path of the assets folder in apk as a string to access the asets files from within the android device with HARMAN ane developed for Play asset delivery?

**Reply by ajwfrost** [reply]
Okay great - I think we may make that case-insensitive, it seems a bit ott to require it to match exactly...

If you're using "install-time" then no, there is no folder that's available for reading assets directly from a path. The Android/Java documentation says we need to use the AssetManager API and there is nothing we've seen for getting a path of the pack or files.

For the other types, then yes you can get the base package path and then resolve from there:
`public function getAssetPackLocation(assetPack:String):String`
(needs for the asset pack to be downloaded before you can get a valid result from this...)

**Reply by mehmury** [reply]
Thanks for Answer..
I will try this in my apps...
